jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from functools import partial
+from typing import Tuple
+
+import jax.numpy as jnp
+import scipy
+from jax import jit
+from jax._src.api import vmap
+from jax._src.numpy.lax_numpy import _check_arraylike
+from jax._src.numpy.util import _wraps
+from jax._src.util import canonicalize_axis
+
+ModeResult = namedtuple('ModeResult', ('mode', 'count'))
+
+@_wraps(scipy.stats.mode)
+@partial(jit, static_argnames=['axis', 'nan_policy'])
+def mode(x: jnp.ndarray, axis: int = 0, nan_policy: str = "propagate") -> ModeResult:
```

Comment:
Let's use `x: ArrayLike` (same import as `Array` above)

Also, you check `axis is None` below, should `axis` be `Optional[int]`?

Reply:
It makes sense
I need to rename the mode input ```x``` to something like ```a``` which is of type ```ArrayLike```
So that I can use ```x = jnp.asarray(a)``` to get ```x``` of type ```Array```
And then I can use methods like ```shape``` or ```ndim``` on ```x```